src/utils.jl

```diff
@@ -44,20 +44,76 @@ vcat_arrays_vector(arrays_vector::AbstractVector{Param}) = vcat(Knet.cat1d.(arra
 """
     reset_minibatch_train!(nlp::AbstractKnetNLPModel)
 
-Select a new training minibatch for `nlp`.
-Typically used before a new evaluation of the loss function/gradient.
+Select a the first training minibatch for `nlp`.
 """
 reset_minibatch_train!(nlp::AbstractKnetNLPModel) =
+  nlp.current_training_minibatch = first(nlp.training_minibatch_iterator)
```

Comment:
I think your method `minibatch_next_train!` should be the default `reset_minibtach_train!`.
You may add a function `first_minibatch_train!`, that you can call it in 
`minibatch_next_train!` when ` i >= nlp.training_minibatch_iterator.imax `.

Reply:
but reset usually mean resetting to the first occurrences or default set, that is why I added that as that

Reply:
Hmm... I will have to update my code then --'.